CMake/CMakeLists.txt

```
@@ -2,6 +2,8 @@
 # List commands don't propagate up, so we have to manually set the parent scope
 # variable after.
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/CMake")
```

```suggestion
list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/CMake" PARENT_SCOPE)
```
I think this can all be done in one line by applying the suggestion and then remove the set() statement after this.

---

Unfortunately no, you can't do this. `list` does not have a mechanism to propagate changes up in scope, see the [last sentence of the Introduction](https://cmake.org/cmake/help/latest/command/list.html#introduction).

---

What you can do is `set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR} ${CMAKE_MODULE_PATH} PARENT_SCOPE)` if you want it to be one line. My original suggestion was wrong.

---

Yes, that you can do. I don't have a strong preference between these, so I'm just going to leave it as is to avoid changing, but thanks for the suggestion.